src/seedu/addressbook/ui/TextUi.java

```diff
@@ -98,71 +73,63 @@ public boolean isCommentLine(String rawInputLine) {
     /**
      * Prompts for the command and reads the text entered by the user.
      * Ignores empty, pure whitespace, and comment lines.
-     *
-     * @see #shouldIgnore(String)
+     * Echos the command back to the user.
      * @return full line entered by the user
      */
     public String getUserCommand() {
         out.print(LINE_PREFIX + "Enter command: ");
         String fullInputLine = in.nextLine();
+
         // silently consume all ignored lines
         while (shouldIgnore(fullInputLine)) {
             fullInputLine = in.nextLine();
         }
-        lastEnteredCommand = fullInputLine;
+
+        showToUser("[Command entered:" + fullInputLine + "]");
         return fullInputLine;
     }
 
-    /**
-     * Retrieves the latest entered non-ignored input line read from the user by {@link #getUserCommand()}
-     */
-    public String getLastEnteredCommand() {
-        return lastEnteredCommand;
-    }
 
-    public void showWelcomeMessage(String version) {
-        showToUser(DIVIDER, DIVIDER, version, MESSAGE_WELCOME, DIVIDER);
+    public void showWelcomeMessage(String version, String storageFilePath) {
+        String storageFileInfo = String.format(MESSAGE_USING_STORAGE_FILE, storageFilePath);
+        showToUser(
+                DIVIDER,
+                DIVIDER,
+                MESSAGE_WELCOME,
+                version,
+                MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE,
+                storageFileInfo,
+                DIVIDER);
     }
 
     public void showGoodbyeMessage() {
         showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
     }
 
-    /**
-     * Echoes the user input back to the user.
-     */
-    public void echoLastEnteredUserCommand() {
-        showToUser("[Command entered:" + lastEnteredCommand + "]");
+
+    public void showInitFailedMessage() {
+        showToUser(MESSAGE_INIT_FAILED, DIVIDER, DIVIDER);
     }
 
     /**
      * Shows message(s) to the user
      */
     public void showToUser(String... message) {
         for (String m : message) {
-            out.println(LINE_PREFIX + m);
+            out.println(LINE_PREFIX + m.replace("\n", LS + LINE_PREFIX));
```

Comment:
I want to limit use of regex to the parser only. Even within parser I would use  it only in some places, just to contrast with the simpler approach. regex is not easy to learn and not essential at early stages.